rest/model/patch.go

```diff
@@ -16,29 +16,36 @@ import (
-	Branch                  *string          `json:"branch"`
-	Githash                 *string          `json:"git_hash"`
-	PatchNumber             int              `json:"patch_number"`
-	Author                  *string          `json:"author"`
-	Version                 *string          `json:"version"`
-	Status                  *string          `json:"status"`
-	CreateTime              *time.Time       `json:"create_time"`
-	StartTime               *time.Time       `json:"start_time"`
-	FinishTime              *time.Time       `json:"finish_time"`
-	Variants                []*string        `json:"builds"`
-	Tasks                   []*string        `json:"tasks"`
-	VariantsTasks           []VariantTask    `json:"variants_tasks"`
-	Activated               bool             `json:"activated"`
-	Alias                   *string          `json:"alias,omitempty"`
-	GithubPatchData         githubPatch      `json:"github_patch_data,omitempty"`
-	ModuleCodeChanges       []APIModulePatch `json:"module_code_changes"`
-	Parameters              []APIParameter   `json:"parameters"`
-	PatchedConfig           *string          `json:"patched_config"`
-	Project                 *string          `json:"project"`
-	CanEnqueueToCommitQueue bool             `json:"can_enqueue_to_commit_queue"`
+	Id                      *string           `json:"patch_id"`
+	Description             *string           `json:"description"`
+	ProjectId               *string           `json:"project_id"`
+	Branch                  *string           `json:"branch"`
+	Githash                 *string           `json:"git_hash"`
+	PatchNumber             int               `json:"patch_number"`
+	Author                  *string           `json:"author"`
+	Version                 *string           `json:"version"`
+	Status                  *string           `json:"status"`
+	CreateTime              *time.Time        `json:"create_time"`
+	StartTime               *time.Time        `json:"start_time"`
+	FinishTime              *time.Time        `json:"finish_time"`
+	Variants                []*string         `json:"builds"`
+	Tasks                   []*string         `json:"tasks"`
+	DownstreamTasks         []DownstreamTasks `json:"downstream_tasks"`
+	VariantsTasks           []VariantTask     `json:"variants_tasks"`
+	Activated               bool              `json:"activated"`
+	Alias                   *string           `json:"alias,omitempty"`
+	GithubPatchData         githubPatch       `json:"github_patch_data,omitempty"`
+	ModuleCodeChanges       []APIModulePatch  `json:"module_code_changes"`
+	Parameters              []APIParameter    `json:"parameters"`
+	PatchedConfig           *string           `json:"patched_config"`
+	Project                 *string           `json:"project"`
+	CanEnqueueToCommitQueue bool              `json:"can_enqueue_to_commit_queue"`
+}
+
+type DownstreamTasks struct {
+	ChildPatchId *string   `json:"child_patch_id"`
```

Comment:
> If we really want to hide the child patch I think we might also need APITask to report the parent as [its version](https://github.com/evergreen-ci/evergreen/blob/5b30a1ec33cad6a30f2bb0bb28ff0f54d3cd445e/rest/model/task.go#L33)

Do you mean that in BuildFromService we should check if a patch is a child patch and if it is, set the version to the parentId? How would that hide it?

Reply:
I agree with the idea to isolate them in a separate `DownstreamTasks` struct. I was just thinking we shouldn't include the `ChildPatchId`.

Reply:
> Do you mean that in BuildFromService we should check if a patch is a child patch and if it is, set the version to the parentId? How would that hide it?

Something like that. If a task is part of a patch we could fetch the patch and if the patch is a child patch then set the version id to the parent id. My thinking is to make the child patches transparent by treating the tasks like they're part of the parent patch and treating the parent patch as if includes that child's tasks.

Reply:
Did you mean if a patch is part of a parent patch then we should fetch the parent?

Reply:
I don't think we'd need to fetch the parent. 
I can give an example: [this task](http://evergreen-staging.spruce.s3-website-us-east-1.amazonaws.com/task/5ffe393097b1d3759dd3c1a3_release_test_release_patch_a2c700f93cae60ded21a2149e4ad7eb6b1b78885_6049134597b1d353b0ea9ce6_21_03_10_19_11_02/logs?execution=0) is part of a child patch. Today, if you get the task through the API the version field will be set to the version for the child patch, in this case the string `"6049134597b1d353b0ea9ce6"`. Since we want to hide the child patch's version we should set the version field to the parent's version, ["6049134597b1d353b0ea9ce5"](http://evergreen-staging.spruce.s3-website-us-east-1.amazonaws.com/version/6049134597b1d353b0ea9ce5/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC).

Reply:
Ah I see. I made some changes, please lmk if I understood correctly.